aiida/orm/utils/mixins.py

```diff
@@ -106,20 +109,66 @@ def _set_function_starting_line_number(self, function_starting_line_number):
         """
         self.base.attributes.set(self.FUNCTION_STARTING_LINE_KEY, function_starting_line_number)
 
+    @property
+    def function_number_of_lines(self):
+        """Return the number of lines of the wrapped function in its source file.
+
+        :returns: the number of lines or None
+        """
+        return self.get_attribute(self.FUNCTION_NUMBER_OF_LINES_KEY, None)
+
+    def _set_function_number_of_lines(self, function_number_of_lines):
+        """Set the number of lines of the wrapped function in its source file.
+
+        :param function_number_of_lines: the number of lines
+        """
+        type_check(function_number_of_lines, int)
+        self.set_attribute(self.FUNCTION_NUMBER_OF_LINES_KEY, function_number_of_lines)
+
     def get_function_source_code(self) -> str | None:
         """Return the source code of the function stored in the repository.
 
         If the source code file does not exist, this will return ``None`` instead. This can happen for example when the
         function was defined in an interactive shell in which case ``store_source_info`` will have failed to retrieve
         the source code using ``inspect.getsourcefile``.
 
+        :returns: The source code of the function or ``None`` if it could not be determined when storing the node.
+        """
+        warn_deprecation('This method will be removed, use `get_function_source_file` instead.', version=3)
+
+        return self.get_function_source_file()
+
+    def get_function_source_file(self) -> str | None:
+        """Return the source code of the function stored in the repository.
+
+        If the source code file does not exist, this will return ``None`` instead. This can happen for example when the
+        function was defined in an interactive shell in which case ``store_source_info`` will have failed to retrieve
+        the source code using ``inspect.getsourcefile``.
+
         :returns: The source code of the function or ``None`` if it could not be determined when storing the node.
         """
         try:
             return self.base.repository.get_object_content(self.FUNCTION_SOURCE_FILE_PATH)
         except FileNotFoundError:
             return None
 
+    def get_function_source_function(self):
```

Comment:
I personally find `get_function_source_function` a bit weird, with the double `function` in there. I would propose `get_source_code_function` and  `get_source_code_file` for the entire file source code. Think that makes more sense no?

Reply:
Yes, I mess up when resolve confliction on github interface. Will change it now.

Reply:
Changed, I hope it is all fine.